### ISSUE TITLE
Unable  to use CMVS with VisualSFM

### ISSUE THREAD

**Issue by Nish160** [pmoulon/CMVS-PMVS#27]
I have downloaded the complete repository containing  precompiled binaries for Win64 and then copied the genOption.exe file to the path containing VisualSFM.exe. But on launching VisualSFM I cant see the option of CMVS dense reconstruction.
Am I missing out something?
Can anyone provide a step wise procedure from downloading CMVS to using it with VisualSFM GUI on windows 64 bit.

Thanks

**Comment by pmoulon** [comment]
Did you follow the installation instruction explained there http://ccwu.me/vsfm/install.html
```
(optional step) Install Yasutaka Furukawa's CMVS software for dense reconstruction.
--
  | - Obtain the CMVS/PMVS binaries. For example, the one by Pierre Moulon.
- Place cmvs/pmvs/genOption/pthreadVc2.dll in the same folder as VisualSFM.exe

```

Else you can still use the command line ;-)

**Comment by Nish160** [comment]
There's no file present at cmvs/pmvs/genOption/pthreadVc2.dll in the zipped folder - CMVS_PMVS-master.zip downloaded from https://github.com/pmoulon/CMVS-PMVS.
It contains the following contents:-
![image](https://user-images.githubusercontent.com/30828701/29201761-d6e8c9c2-7e7e-11e7-965d-00e2ad1523da.png)

So, what I did is, I copied the files present in  CMVS-PMVS-master\binariesWin-Linux\Win64-VS2010   to the location whereVisualSFM.exe is present.

![image](https://user-images.githubusercontent.com/30828701/29202840-79a751b0-7e89-11e7-88a2-2c3152b622a9.png)

**Comment by luotinen** [comment]
The button appears after coarse reconstruction. Please use a small set of photographs and run sift+matching+coarse and the button will appear.

**Comment by pmoulon** [comment]
@Nish160 Sorry we removed the pthread dependency. That's why the dll is not longer required.

By the way you can also switch to other 3d reconstruction framework (OpenMVG + OpenMVS/MVE)

**Comment by Darker** [comment]
Why is this closed. The issue is not resolved.

**Comment by pmoulon** [comment]
Answers have been provided.
After 2 months without an answer from the issuer, I closed the issue.

**Comment by pa-sy** [comment]
If the person who asked the question does not understand the answer or does not think the question has been answered then it has not been answered satisfactorily and it should not be closed. You have assumed far too mauch knowledge about how this works for your answer to be helpful to anyone who does not work on  this code themselves. In which case they would not have needed to post the question. Please can you reopen because reading this as a software developer I do not have enough information to understand your answer let alone someone who is not a developer.

**Comment by ahananq** [comment]
Hi,

I have used VisualSFM with CMVS and it was working smooth. Now I am getting this error. Can you guide ?

when I run the "Dense Reconstruction" ,it showed the "ERROR: the above CMVS command failed!



Running Yasutaka Furukawa's CMVS tool...
cmvs "D:\VisualSFM\Data.Input\Exp#1\2.nvm.cmvs\01\\" 50 12
ERROR: the above CMVS command failed!
FAQ: http://ccwu.me/vsfm/doc.html#errors

**Comment by pmoulon** [comment]
Hello such error often appears from an invalid command-line call about the path.
Try the path without the `\` at the end

**Comment by ahananq** [comment]
Hi thank you for quick response.
actually I am using VisualSFM on window, and using provided binaries of CMVS, PMVS2, genOption & pthreadVc2.dll copied in same folder as VisualSFM.exe.
I downloaded all the files again to avoid corrupted files. I am not a programmer, I want to use photogrammetry for some project. I have followed instruction provided in FAQ regarding this issue but unable to resolve this issue.

Regards
